lib/eunit/src/eunit_lib.erl

```diff
@@ -574,6 +574,17 @@ trie_match([], []) ->
 trie_match([], _T) ->
     no.
 
+%% ---------------------------------------------------------------------
+%% Convert a string of unknown encoding to a binary
+
+try_to_binary(S) when is_list(S) ->
```

Comment:
If the source file is latin-1 and contains characters above 128, does the text automagically converted to unicode anyway?

Like I admitted before, I got to this implementation by trial and error. I checked files and output with a hex viewer to not let the console's encoding disturb the comparison, then created tests that mirror these results. It's possible that I missed some cases.

Given that I agree that the result is difficult to understand, I am considering going through the eunit code again and unify the input/output so that it's easier to grasp.